Src/FluentAssertions/Numeric/ComparableTypeAssertions.cs

```diff
@@ -35,7 +36,7 @@ public ComparableTypeAssertions(IComparable<T> value) : base(value)
         public AndConstraint<ComparableTypeAssertions<T>> Be(T expected, string because = "", params object[] becauseArgs)
         {
             Execute.Assertion
-                .ForCondition(ReferenceEquals(Subject, expected) || (Subject.CompareTo(expected) == Equal))
+                .ForCondition(ReferenceEquals(Subject, expected) || ((Subject.CompareTo(expected) == Equal) && Subject.IsSameOrEqualTo(expected)))
```

Comment:
I think we should _only_ use `ISameOrEqual` since that will ensure `Be` behaves like other methods

Reply:
Do I understand correctly that the easiest solution then would be to just remove these methods `Be` and `NotBe` altogether from `ComparableTypeAssertions`? It would fall back on the ObjectAssertions. This would have the downside, though, of not checking the implementation of `CompareTo` anymore.

Reply:
Yep. That's why we [suggested](https://github.com/fluentassertions/fluentassertions/issues/172#issuecomment-59910838) adding another method for that.

Reply:
From the thumbs up on my suggestion I (wrongly) assumed that was the way to go. Maybe I jumped the gun on this.

Reply:
You're almost there, aren't you?

Reply:
Now I am confused. My understanding is that you only want Equality and no Reference Equality and Comparison checks?
What I implemented is checking all three, to make sure that people who are depending on the behaviour that the comparison is checked is not implicitly removed in a new version. Most of the implementation is checking these three cases.

Reply:
Reference quality is usually checked with `BeSameAs`.